src/OversetSimulation.cpp

```diff
@@ -192,6 +177,83 @@ bool OversetSimulation::do_connectivity(const int tstep)
     return (tstep > 0) && (tstep % m_overset_update_interval) == 0;
 }
 
+void OversetSimulation::print_timing(const int nt)
+{
+    // overall timestep timing
+    auto timing_summary = m_timers_exa.get_timings_summary(
+        "Exawind", nt, m_comm, m_printer.io_rank());
+    m_printer.echo(timing_summary);
+
+    auto timing_detail = m_timers_exa.get_timings_detail(
+        "Exawind", nt, m_comm, m_printer.io_rank());
+    m_printer.timing_to_file(timing_detail);
+
+    MPI_Barrier(m_comm);
+
+    // tioga timing
+    timing_summary = m_timers_tg.get_timings_summary(
+        "Tioga", nt, m_comm, m_printer.io_rank());
+    m_printer.echo(timing_summary);
+
+    timing_detail = m_timers_tg.get_timings_detail(
+        "Tioga", nt, m_comm, m_printer.io_rank());
+    m_printer.timing_to_file(timing_detail);
+
+    // cfd solver-specific timing
+    for (auto& ss : m_solvers) {
+        MPI_Barrier(m_comm);
+
+        ParallelPrinter printer(ss->comm());
+
+        // summary timings
+        if (ss->is_amr()) {
+            std::string timings_summary = ss->m_timers.get_timings_summary(
+                ss->identifier(), nt, ss->comm(), printer.io_rank());
+            printer.echo(timings_summary);
+        } else { // logic to add and average total time
+            std::vector<double> total_send(3, 0.0);
+            ss->m_timers.total_times(
+                total_send[0], total_send[1], total_send[2], ss->comm(),
+                printer.io_rank());
+
+            if (printer.is_io_rank()) {
+                MPI_Send(total_send.data(), 3, MPI_DOUBLE, 0, 0, m_comm);
+            }
+        }
+
+        // detailed timings
+        std::string timings_detail = ss->m_timers.get_timings_detail(
+            ss->identifier(), nt, ss->comm(), printer.io_rank());
+        printer.timing_to_file(timings_detail);
+    }
+
+    MPI_Barrier(m_comm);
+
+    // average nalu-wind total timings and print
+    if (m_printer.is_io_rank()) {
+        double nalu_total_min, nalu_total_avg, nalu_total_max;
+        for (int i = 0; i < m_nw_start_rank.size(); ++i) {
```

Comment:
Since you only use `i` to index `start_rank` you can just `for(const auto& start_rank: m_nw_start_rank` and use `start_rank`